src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -187,53 +187,43 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         const preferredSearchToken = new CancellationTokenSource();
         this.findPreferredInProgress.set(document, preferredSearchToken);
 
-        if (!this.isLocalLaunch) {
-            // For a remote connection check for new live kernel models before we find preferred
-            this.updateRemoteConnections(preferredSearchToken.token)
-                .then(() => {
-                    this.setController(document, preferredSearchToken.token).catch((error) => {
-                        traceError(error);
-                    });
-                })
-                .finally(() => {
-                    // Make sure that we clear our finding in progress when done
-                    this.findPreferredInProgress.delete(document);
-                });
-        } else {
-            this.setController(document, preferredSearchToken.token).finally(() => {
-                // Make sure that we clear our finding in progress when done
-                this.findPreferredInProgress.delete(document);
-            });
-        }
-    }
-
-    // Set the controller for this notebook document
-    private async setController(document: NotebookDocument, cancelToken: CancellationToken) {
```

Comment:
This method doesn't set the controller at all, if anything it finds the preferred controller and sets the preferred controller.
We also have an existing method named `setPreferredController`.
Also, it wasn't clear to me my we were clearing ` this.controllerMapping.delete(document);` in line `214`.

I felt that it makes more sense to have this code inline in the event handler. 
To me the code reads better, but if others disagree please reject this PR

**Again, this is NOT a required PR, just a simple codehealth/refactor.**

Reply:
Looks good to me. Should probably wait for Ian as he wrote this originally.

Reply:
The main reason that this function was structured as it is, is that I believe we discussed wanted to keep the onDidOpen / onDidClose sync and not have awaits in them. However I can't track down that initial discussion on that. If aync on the doc functions is not a problem I'd prefer this as well.

Reply:
Looks like we're ok, please can we get this approved.